pylint/test/input/func_e0303.py

```diff
@@ -0,0 +1,44 @@
+"""Check bad value returned by __len__ """
+
+# pylint: disable=too-few-public-methods, missing-docstring, no-self-use
+import sys
+
+import six
+
+
+class FirstGoodLen(object):
+    """__len__ returns <type 'int'>"""
+
+    def __len__(self):
+        return 0
+
+
+class SecondGoodLen(object):
+    """__len__ returns <type 'long'>"""
+
+    def __len__(self):
+        return sys.maxsize + 1
+
+
+class LenMetaclass(type):
+    def __len__(cls):
+        return 1
+
+
+@six.add_metaclass(LenMetaclass)
+class ThirdGoodLen(object):
+    """Length through the metaclass."""
+
+
+class FirstBadLen(object):
+    """ __len__ returns a negative integer """
+
+    def __len__(self):
+        return -1
+
+
+class SecondBadLength(object):
+    """ __len__ float """
+
+    def __len__(self):
+        return 3.0
```

Comment:
I'll work on it. This case we're caught by some already existing unittests (that's why check for `Uninferable` was introduced), but I understand it should be explicitly stated in this feature test suite.